core/src/test/scala/org/apache/spark/ExternalShuffleServiceSuite.scala

```diff
@@ -67,7 +67,9 @@ class ExternalShuffleServiceSuite extends ShuffleSuite with BeforeAndAfterAll wi
 
   // This test ensures that the external shuffle service is actually in use for the other tests.
   test("using external shuffle service") {
-    sc = new SparkContext("local-cluster[2,1,1024]", "test", conf)
+    val confWithoutHostLocalRead =
+      conf.clone.set(config.SHUFFLE_HOST_LOCAL_DISK_READING_ENABLED, false)
```

Comment:
this test case makes me think that mayeb we should disable this configuration just for local-cluster mode.  That's a testing util intended to feel a lot like a real cluster -- I think it would be good to still have it be going over the network (unless the user explicitly sets it to true).

Makes the defaulting of this option a little more complex, but seems worth it to me.